src/amo/components/PermissionsCard/permissions.js

```diff
@@ -0,0 +1,237 @@
+      clipboardRead: i18n.gettext('Get data from the clipboard'),
+      clipboardWrite: i18n.gettext('Input data to the clipboard'),
+      devtools: i18n.gettext('Extend developer tools to access your data in open tabs'),
+      downloads: i18n.gettext('Download files and read and modify the browser’s download history'),
+      'downloads.open': i18n.gettext('Open files downloaded to your computer'),
+      find: i18n.gettext('Read the text of all open tabs'),
+      geolocation: i18n.gettext('Access your location'),
+      history: i18n.gettext('Access browsing history'),
+      management: i18n.gettext('Monitor extension usage and manage themes'),
+      // In Firefox the following message replaces the name "Firefox" with the
+      // current brand name, e.g., "Nightly", but we do not need to do that.
+      nativeMessaging: i18n.gettext('Exchange messages with programs other than Firefox'),
+      notifications: i18n.gettext('Display notifications to you'),
+      pkcs11: i18n.gettext('Provide cryptographic authentication services'),
+      proxy: i18n.gettext('Control browser proxy setting'),
+      privacy: i18n.gettext('Read and modify privacy settings'),
+      sessions: i18n.gettext('Access recently closed tabs'),
+      tabs: i18n.gettext('Access browser tabs'),
+      tabHide: i18n.gettext('Hide and show browser tabs'),
+      topSites: i18n.gettext('Access browsing history'),
+      unlimitedStorage: i18n.gettext('Store unlimited amount of client-side data'),
+      webNavigation: i18n.gettext('Access browser activity during navigation'),
+      allUrls: i18n.gettext('Access your data for all websites'),
+      wildcard: (domain) => {
+        return i18n.sprintf(
+          i18n.gettext('Access your data for sites in the %(domain)s domain'),
+          { domain }
+        );
+      },
+      // Note that we do not need to use ngettext for this as count will never
+      // be less than 2.
+      tooManyWildcards: (count) => {
+        return i18n.sprintf(
+          i18n.gettext('Access your data in %(count)s other domains'),
+          { count: i18n.formatNumber(count) },
+        );
+      },
+      oneSite: (site) => {
+        return i18n.sprintf(
+          i18n.gettext('Access your data for %(site)s'),
+          { site }
+        );
+      },
+      // Note that we do not need to use ngettext for this as count will never
+      // be less than 2.
+      tooManySites: (count) => {
+        return i18n.sprintf(
+          i18n.gettext('Access your data on %(count)s other sites'),
+          { count: i18n.formatNumber(count) },
+        );
+      },
+    };
+  }
+
+  // Get a list of permissions from the correct platform file.
+  getCurrentPermissions(
+    addon: AddonType,
+    userAgentInfo: UserAgentInfoType
+  ): Array<string> {
+    const agentOsName =
+      userAgentInfo.os.name && userAgentInfo.os.name.toLowerCase();
+    const platform = userAgentOSToPlatform[agentOsName];
+
+    if (!platform) {
```

Comment:
True, but I think I need to bring back `fakeUserAgent` from https://github.com/mozilla/addons-frontend/pull/4462/commits/07f444f81843b6d6bf5f63205490230180047887#diff-76f7a688842283b52b432e415dcf2e3eR42, or for me to just create a one time object to pass into `getCurrentPermissions` because I don't think we can get an invalid platform from `UAParser`, correct? I guess if I only need it for the one test I can just create an object for that purpose.

Reply:
I agree that testing the log message is unhelpful. It's good enough to execute the statements within the if branch -- that will make sure no errors are thrown.